packages/launch/src/focused-launch/plan-details/index.tsx

```diff
@@ -3,22 +3,80 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import { Link } from 'react-router-dom';
+import * as React from 'react';
```

Comment:
Out of curiosity, what are the benefits of `import * as React` over `import React` ?

Reply:
I thought this. Looks like it's being used for the functional component typings

Reply:
I thought it would work anyway

Reply:
I know we used this way of importing for a while for performance reasons. I believe now we have webpack configured so this isn't necessary anymore. Pinging @sirreal to confirm though 🙏

Reply:
If you do `import React...` you need `allowSyntheticDefaultImports = true` in tsconfig. `import * as React` [is more semantically accurate and makes TypeScript happy](https://stackoverflow.com/a/55285991/1247990). But don't quote me on this. I think (not sure) I saw @sirreal do it, and he is my TypeScript prophet.

Reply:
There were some perf concerns and I don't know whether they're relevant. Either way, I think we have conclusive evidence that `import * as React from 'react';` is "right": https://github.com/facebook/react/pull/18102

> [`import * as React from "react";`] is the correct way to import React from an ES module since the ES module will not have a default export. Only named exports.
> 
> I'm doing some changes to named exports and this will stop working. For open source ES modules we'll probably publish some default object that warns as an upgrade path but we can't use that in our own code.

We could add a lint rule for this and/or codemod it, but I don't know whether there is significant impact beyond consistency.

Reply:
Thank you for the explanation! I learnt something new

Reply:
> If you do `import React...` you need `allowSyntheticDefaultImports = true` in tsconfig. `import * as React` [is more semantically accurate and makes TypeScript happy](https://stackoverflow.com/a/55285991/1247990).

Yes, I think this is right. Note, this isn't really about TypeScript but about ECMAScript modules which are quirky. I wrote a bit here (p4TIVU-8Lf-p2).